src/login/logind-button.c

```diff
@@ -275,6 +312,35 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
                                 manager_handle_action(b->manager, INHIBIT_HANDLE_HIBERNATE_KEY, b->manager->handle_hibernate_key, b->manager->hibernate_key_ignore_inhibited, true);
                         }
                         break;
+
+                case KEY_ESC:
+                        if (b->manager->handle_secure_attention_key_press != HANDLE_IGNORE) {
+                                if (b->mods_depressed == (Button_Modifier_Ctrl | Button_Modifier_Alt | Button_Modifier_Shift)) {
+                                        log_debug("Secure Attention Key sequence pressed short.");
+                                        secure_attention_key_handler(b);
+                                }
+                        }
+                        break;
+
+                case KEY_LEFTSHIFT:
+                case KEY_RIGHTSHIFT:
+                        b->mods_depressed |= Button_Modifier_Shift;
```

Comment:
this doesn't look right, it means that if i press both shift left and shift right at the same time, and then release one, that it is assumed that shift is not pressed anymore.

i.e. please track the states of left/right separately, and then check both when actually looking for the key combo